18F/identity-idp
# Revert from Propshaft back to Sprockets

## 🛠 Summary of changes

This reverts the changes that migrated from Sprockets to Propshaft (#8387, #8402, #8409) due to issues observed in deployed environments.

Specifically, there seems to be some unpredictable behavior related to our use of CDNs via `asset_host`, where recent changes in Propshaft result in assets resolving to invalid paths (e.g. `/assets/login.gov/assets/angle-arrow-up-white-[digest].svg` vs. expected `/assets/angle-arrow-up-white-[digest].svg`). The idea with reverting is to allow for this behavior to stabilize in Propshaft.

This does preserve some of the work in the reverted pull requests, notably spec revisions and other improvements which are valuable regardless of Sprockets vs. Prosphaft. These are reflected in commits 857112cf62b1db7dc218a4a336c440d4a144e9a9, 080636a382e2acead74a751c16065cb81a5bb77d, and 76d226860a8b18ed13c8dbf39be6a0b4c47064f2.

Related Slack discussions:
- https://gsa-tts.slack.com/archives/C0NGESUN5/p1684248814878349
- https://gsa-tts.slack.com/archives/C0NGESUN5/p1684328100876409
- https://gsa-tts.slack.com/archives/C0NGESUN5/p1684269870234819

Related upstream issues / commits:
- https://github.com/rails/propshaft/pull/131
   - This may be a fix for the problematic behavior observed in the (current-as-of-writing) v0.7.0
- https://github.com/rails/propshaft/issues/104
- https://github.com/rails/propshaft/pull/118

## 📜 Testing Plan

1. Go to http://localhost:3000
2. Observe that images and styles appear as expected
